fabric_cf/orchestrator/core/orchestrator_handler.py

```diff
@@ -390,4 +393,80 @@ def get_slice_graph(self, *, token: str, slice_id: str) -> dict:
         except Exception as e:
             self.logger.error(traceback.format_exc())
             self.logger.error(f"Exception occurred processing get_slice_graph e: {e}")
-            raise e
\ No newline at end of file
+            raise e
+
+    def renew_slice(self, *, token: str, slice_id: str, new_lease_end_time: str):
+        """
+        Renew a slice
+        :param token Fabric Identity Token
+        :param slice_id Slice Id
+        :param new_lease_end_time: New Lease End Time in UTC in '%Y-%m-%d %H:%M:%S' format
+=       :raises Raises an exception in case of failure
+        :return:
+        """
+        failed_to_extend_rid_list = []
+        try:
+            new_end_time = datetime.strptime(new_lease_end_time, Constants.RENEW_TIME_FORMAT)
```

Comment:
Are there any exceptions from this? This can't always succeed...

Reply:
Oh never mind, this is inside a try block and error in the end. How is the exception communicated back to user though?

Reply:
API fails with 500 error and exception string is returned. I'll change it to return Bad Request error

Reply:
OK